osu.Game/Online/API/APIAccess.cs

```diff
@@ -309,7 +309,7 @@ private bool handleRequest(APIRequest req)
                 if (IsLoggedIn) state.Value = APIState.Online;
 
                 failureCount = 0;
-                return true;
+                return req.CompletionState == APIRequestCompletionState.Completed;
```

Comment:
Shouldn't this return early if `!Completed` before the state and failure count are set?

Reply:
To avoid resetting `failureCount`?

Reply:
That and to not set the online state, but maybe that's not an issue because if the request fails we won't have a user ID? Not sure about the case when the current state is `Failing` and this request fails, that then changes the API into an "Online" state.

Reply:
I've moved it, we'll see how it works. All this logic still needs a good rewrite at some point.